### PR TITLE
Added swashbuckle/swagger gen to project see here for docs - https://…

### PR DESCRIPTION
**Pull request by tombiddulph**
- Compared the 2 obvious options Nswag & Swashbuckle, they both seem to do the same thing. I ended up choosing swashbuckle as it seems like a more mature project.

- Setup swashbuckle with a basic configuration
- Annotated the ExampleController as an example of what needs to be done to generate the swagger data
- Enabled the 'XML' comment option to enrich the generated swagger data with in code xml comments, any methods/classes that aren't documented will generate a compiler warning which we should disable/reenable when appropriate.
- Set the path for the swagger docs to be /api-docs

- The info object needs reviewing, should we even include it?

```options.SwaggerDoc("v1", new Info
                {
                    Title = "Open Banking Service",
                    Version = "V1",
                    Contact = new Contact
                    {
                        Email = "hello@example.com",
                        Name = "Hello Developer"
                    },
                    License = new License
                    {
                        Name = "License name",
                        Url = "http://google.com"
                    }
                });```

### CODE REVIEW

File: src/Api/Api.csproj [hunk 1]
```diff
@@ -7,10 +7,15 @@
     <UserSecretsId>3bdd0553-9b4b-4d58-9c3f-9b5bc6e2d6bb</UserSecretsId>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DocumentationFile>C:\Users\Thomas Biddulph\source\repos\open-banking-service\src\Api\Api.xml</DocumentationFile>
```
Comment (hwarrior) [inline src/Api/Api.csproj]: This local path probably needs to be updated:)

File: src/Api/Startup.cs [hunk 2]
```diff
@@ -1,52 +1,92 @@
...
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
 using SmartRequest.OpenBankingService.Domain.Example;
 using SmartRequest.OpenBankingService.Infrastructure.Repositories;
+using Swashbuckle.AspNetCore.Swagger;
 
+#pragma warning disable 1591
```
Comment (hwarrior) [inline src/Api/Startup.cs]: Just for my benefit as I don't know - what does this do?

File: src/Domain/Domain.csproj [hunk 3]
```diff
@@ -4,4 +4,8 @@
     <TargetFramework>netcoreapp2.2</TargetFramework>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DocumentationFile>C:\Users\Thomas Biddulph\source\repos\open-banking-service\src\Domain\Domain.xml</DocumentationFile>
```
Comment (hwarrior) [inline src/Domain/Domain.csproj]: Local path here too.

**Comment by tombiddulph** [comment]
`#pragma warning disable 1591` disables the Missing xml comment compiler warning.

The reason that It's enabled is to use xml comments within the code to enrich the swagger documentation.

Alternative we could leave the setting enabled and disable the warning project wide 

```<PropertyGroup>
  <GenerateDocumentationFile>true</GenerateDocumentationFile>
  <NoWarn>$(NoWarn);1591</NoWarn>
</PropertyGroup>```

**Pull request merged**